### DISCUSSION TITLE
[Question] Understanding the Receiver of set_attr in Scrollable Controls and Scroll Animation Handling

### DISCUSSION THREAD

**Discussion by jaluscg** [flet-dev/flet#2392] [Q&A]
### Question

I've been exploring the behavior of `set_attr` in Scrollable controls and encountered some behavior I'm hoping to understand better.

Firstly, I noticed that for a `Scrollable` control, `set_attr` seems to be assigned like this:

```python
self._set_attr: <bound method Control._set_attr of SpecificColumn(height=202.33333333333334, scroll='always')>
```

Attempting to add a maxse attribute to this dictionary to expand the scroll functionality didn't seem to have any effect:

```python
self._set_attr: <bound method Control._set_attr of SpecificColumn(height=202.33333333333334, scroll='always', maxse=2000)>
```

Furthermore, modifying e.data to change maxse to, for example, 2000, also did not yield any changes:

```python
self.__result_converter <function SpecificScrollableControl.__init__.<locals>.convert_on_scroll_event_data at 0x7f29843040e0>
e.data in sync_handler: {"p":23,"minse":0,"maxse":2000,"vd":683,"t":"start"}
e.data = {"p":23,"minse":0,"maxse":2000,"vd":683,"t":"start"} 
```

My question is: Which part of the code or document takes this dictionary and converts the "on scroll True" into an animation that allows for scrolling and applies rules? I'm trying to understand the underlying mechanism that controls the scroll behavior and animation in these specific controls.

Any guidance or pointers to the relevant sections of the documentation or codebase would be greatly appreciated.

Thank you!

**Comment by jaluscg** [comment]
These are the attributes I found, but I'm not sure which of these to modify in order to change the 'maxse' of the scroll or to alter its behavior:

self.__on_scroll<user_controls.for_specific_table.SpecificEventHandler.SpecificEventHandler object at 0x7f52c0c271d0>
ejecutando @scroll.setter def scroll if isinstance
Attribute '_ConstrainedControl__animate_offset': Unknown
Attribute '_ConstrainedControl__animate_opacity': Unknown
Attribute '_ConstrainedControl__animate_position': Unknown
Attribute '_ConstrainedControl__animate_rotation': Unknown
Attribute '_ConstrainedControl__animate_scale': Unknown
Attribute '_ConstrainedControl__animate_size': Unknown
Attribute '_ConstrainedControl__offset': Unknown
Attribute '_ConstrainedControl__rotate': Unknown
Attribute '_ConstrainedControl__scale': Unknown
Attribute '_Control__attrs': Unknown
Attribute '_Control__col': Unknown
Attribute '_Control__data': Unknown
Attribute '_Control__event_handlers': Unknown
Attribute '_Control__expand': Unknown
Attribute '_Control__page': Unknown
Attribute '_Control__previous_children': Unknown
Attribute '_Control__uid': Unknown
Attribute '_SpecificColumn__set_alignment': Unknown
Attribute '_SpecificColumn__set_horizontal_alignment': Unknown
Attribute '_SpecificScrollableControl__on_scroll': Unknown
Attribute '_SpecificScrollableControl__scroll': Unknown
Attribute '_SpecificScrollableControl__set_scroll': Unknown
Attribute '__class__': Unknown
Attribute '__delattr__': Unknown
Attribute '__dict__': Unknown
Attribute '__dir__': Unknown
Attribute '__doc__': Unknown
Attribute '__eq__': Unknown
Attribute '__format__': Unknown
Attribute '__ge__': Unknown
Attribute '__getattribute__': Unknown
Attribute '__getstate__': Unknown
Attribute '__gt__': Unknown
Attribute '__hash__': Unknown
Attribute '__init__': Unknown
Attribute '__init_subclass__': Unknown
Attribute '__le__': Unknown
Attribute '__lt__': Unknown
Attribute '__module__': Unknown
Attribute '__ne__': Unknown
Attribute '__new__': Unknown
Attribute '__reduce__': Unknown
Attribute '__reduce_ex__': Unknown
Attribute '__repr__': Unknown
Attribute '__setattr__': Unknown
Attribute '__sizeof__': Unknown
Attribute '__str__': Unknown
Attribute '__subclasshook__': Unknown
Attribute '__weakref__': Unknown
Attribute '_add_event_handler': Unknown
Attribute '_before_build_command': Unknown
Attribute '_build': Unknown
Attribute '_build_add_commands': Unknown
Attribute '_build_command': Unknown
Attribute '_convert_attr_json': Unknown
Attribute '_dispose': Unknown
Attribute '_get_attr': Unknown
Attribute '_get_children': Unknown
Attribute '_get_control_name': Unknown
Attribute '_get_event_handler': Unknown
Attribute '_get_value_or_list_attr': Unknown
Attribute '_id': Unknown
Attribute '_is_isolated': Unknown
Attribute '_previous_children': Unknown
Attribute '_remove_control_recursively': Unknown
Attribute '_set_attr': Unknown
Attribute '_set_attr_internal': Unknown
Attribute '_set_attr_json': Unknown
Attribute '_set_value_or_list_attr': Unknown
Attribute '_wrap_attr_dict': Unknown
Attribute 'alignment': Unknown
Attribute 'animate_offset': Unknown
Attribute 'animate_opacity': Unknown
Attribute 'animate_position': Unknown
Attribute 'animate_rotation': Unknown
Attribute 'animate_scale': Unknown
Attribute 'animate_size': Unknown
Attribute 'aspect_ratio': Unknown
Attribute 'auto_scroll': Unknown
Attribute 'bottom': Unknown
Attribute 'build_update_commands': Unknown
Attribute 'clean': Unknown
Attribute 'clean_async': Unknown
Attribute 'col': Unknown
Attribute 'controls': Unknown
Attribute 'copy_attrs': Unknown
Attribute 'data': Unknown
Attribute 'did_mount': Unknown
Attribute 'did_mount_async': Unknown
Attribute 'disabled': Unknown
Attribute 'event_handlers': Unknown
Attribute 'expand': Unknown
Attribute 'height': 202.33333333333334
Attribute 'horizontal_alignment': Unknown
Attribute 'key': Unknown
Attribute 'left': Unknown
Attribute 'offset': Unknown
Attribute 'on_animation_end': Unknown
Attribute 'on_scroll': Unknown
Attribute 'on_scroll_interval': Unknown
Attribute 'opacity': Unknown
Attribute 'page': Unknown
Attribute 'reverse': Unknown
Attribute 'right': Unknown
Attribute 'rotate': Unknown
Attribute 'run_spacing': Unknown
Attribute 'scale': Unknown
Attribute 'scroll': Unknown
Attribute 'scroll_to': Unknown
Attribute 'scroll_to_async': Unknown
Attribute 'spacing': Unknown
Attribute 'tight': Unknown
Attribute 'tooltip': Unknown
Attribute 'top': Unknown
Attribute 'uid': Unknown
Attribute 'update': Unknown
Attribute 'update_async': Unknown
Attribute 'visible': Unknown
Attribute 'width': Unknown
Attribute 'will_unmount': Unknown
Attribute 'will_unmount_async': Unknown
Attribute 'wrap': Unknown

**Reply by ndonkoHenri** [reply]
What is `maxse`?

**Reply by jaluscg** [reply]
"The maxse corresponds to max_scroll_extent, which is the maximum width allowed for the scroll animation. When initially set up, the e.data generates a dictionary containing this information (e.g., e.data = {"p":0, "minse":0, "maxse":817, "vd":683, "t":"user", "dir":"reverse"}). As scrolling occurs, the p (pixels) progresses through the available pixel range from minse to maxse. The issue I'm encountering is that even when I try to modify this value, for instance, by changing the maxse to a different value, the scrollbar behaves as if no change has been made."
